docs/public/architecture/components/fx.md

```diff
@@ -0,0 +1,194 @@
+# Overview of Fx
+
+The Agent uses [Fx](https://uber-go.github.io/fx) as its application framework. While the linked Fx documentation is thorough, it can be a bit difficult to get started with. This document describes how Fx is used within the Agent in a more approachable style.
+
+## What Is It?
+
+Fx's core functionality is to create instances of required types "automatically," also known as [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection). Within the agent, these instances are components, so Fx connects components to one another. Fx creates a single instance of each component, on demand.
+
+This means that each component declares a few things about itself to Fx, including the other components it depends on. An "app" then declares the components it contains to Fx, and instructs Fx to start up the whole assembly.
+
+## Providing and Requiring
+
+Fx connects components using types. Within the Agent, these are typically interfaces named `Component`. For example, `scrubber.Component` might be an interface defining functionality for scrubbing passwords from data structures:
+
+=== ":octicons-file-code-16: scrubber/component.go"
```

Comment:
this will only be visible on github pages, correct?

Reply:
Yes that's right